plugins/diggingInvaders/diggingInvaders.cpp

```diff
@@ -372,8 +372,6 @@ void findAndAssignInvasionJob(color_ostream& out, void* tickTime) {
         clearDijkstra();
         return;
     }
-    EventManager::unregister(EventManager::EventType::TICK, findJobTickHandler, plugin_self);
-    EventManager::registerTick(findJobTickHandler, 1, plugin_self);
```

Comment:
Why was this removed? I see nowhere else where `findJobTickHandler` is registered, so I suspect this is disabling the handler entirely.

Reply:
Indeed. I assumed it must be registered somewhere else otherwise it wouldn't ever be registered.. but obviously it gets called automaticalled, or manually.. in order to register initially. This seems like a good enough reason to provide both mechanisms. I'll put these lines back, and ensure `registerTick` doesn't result in re-queued events.

Reply:
I think the `unregister()` call ensured that the event was unregistered (before it was re-registered), if it was registered already. Not sure what would have happened with just one `registerTick` call previously, though.

If we can preserve the old behavior where (I assume) this didn't result in the same handler getting called an increasing number of times per tick, I agree that that would likely simplify things on the caller side.

Reply:
This should function as it did before now.

`registerTick` gets called when the function executes at least once. `registerTick` will not add the handler to the list, but does queue the event. `manageTick` calls the event when the tick happens, removes it from the queue, adds it to a removal set.. then later after all the events have fired the events in the removal set get cross referenced against the handlers list, and if it is located it gets removed from the list and re-registered with `registerListener` (which **does** add event handlers to the handlers list). The flip side being that if no event handler is found the event isn't re-registered, as will be the case with `registerTick` tick events.